src/util.js

```diff
@@ -114,8 +115,10 @@ u.boolean = function(s) {
   return s == null || s === '' ? null : s==='false' ? false : !!s;
 };
 
-u.date = function(s) {
-  return s == null || s === '' ? null : Date.parse(s);
+// parse a date with optional format
+u.date = function(s, format) {
+  var d = format ? d3_timeF.format(format) : Date;
```

Comment:
This code will allocate a new formatter instance for **every** value we parse. That's a performance and garbage collection no-no. Instead, we might pass a formatter instance (not format string) as input: allocate the formatter _once_ within the parsing routines, then pass it in here. We could optionally support a string argument as well, but would then need to perform some run-time type checking here. However, at that point it might make more sense to just call the formatter directly in the parsing code to avoid all the overhead.